SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriverRPC.cpp

```diff
@@ -381,12 +381,15 @@ bool FSpatialNetDriverRPC::ApplyRPC(Worker_EntityId EntityId, const FRPCPayload&
 	if (TargetObject == nullptr)
 	{
 		const TWeakObjectPtr<UObject> ActorReceivingRPC = NetDriver.PackageMap->GetObjectFromEntityId(EntityId);
-		AActor* Actor = CastChecked<AActor>(ActorReceivingRPC.Get());
-		checkf(Actor != nullptr, TEXT("Receiving actor should have been checked in CanReceiveRPC"));
-		UE_LOG(LogSpatialNetDriverRPC, Error,
-			   TEXT("Failed to execute RPC on Actor %s (Entity %llu)'s Subobject %i because the Subobject is null"), *Actor->GetName(),
-			   EntityId, RPCData.Offset);
-
+		if (ActorReceivingRPC.Get())
+		{
+			AActor* Actor = CastChecked<AActor>(ActorReceivingRPC.Get());
+			checkf(Actor != nullptr, TEXT("Receiving actor should have been checked in CanExtractRPC"));
+			UE_LOG(LogSpatialNetDriverRPC, Error,
+				   TEXT("Failed to execute RPC on Actor %s (Entity %llu)'s Subobject %i because the Subobject is null"), *Actor->GetName(),
+				   EntityId, RPCData.Offset);
+		}
+		
```

Comment:
```suggestion
		if (UObject* ActorRecieving = ActorReceivingRPC.Get())
		{
			AActor* Actor = CastChecked<AActor>(ActorRecieving);
			checkf(Actor != nullptr, TEXT("Receiving actor should have been checked in CanExtractRPC"));
			UE_LOG(LogSpatialNetDriverRPC, Error,
				   TEXT("Failed to execute RPC on Actor %s (Entity %llu)'s Subobject %i because the Subobject is null"), *Actor->GetName(),
				   EntityId, RPCData.Offset);   
		}
		else
		{
			UE_LOG(LogSpatialNetDriverRPC, Log,
				   TEXT("Actor destroyed before RPC could execute, Entity %llu"), EntityId);
		}
		
```

Reply:
Thank you, Simon, limiting the scope and adding a log make sense to me, I should have done this from the beginning :D